scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyUri.scala

```diff
@@ -39,32 +38,32 @@ import scala.collection.JavaConverters._
 trait SparkeyUri extends Serializable {
   val basePath: String
   def getReader: SparkeyReader
+
   private[sparkey] def exists: Boolean
   override def toString: String = basePath
 }
 
 private[sparkey] object SparkeyUri {
   def apply(basePath: String, opts: PipelineOptions): SparkeyUri =
     if (ScioUtil.isLocalUri(new URI(basePath))) {
-      new LocalSparkeyUri(basePath)
+      LocalSparkeyUri(basePath)
     } else {
       new RemoteSparkeyUri(basePath, opts)
     }
   def extensions: Seq[String] = Seq(".spi", ".spl")
-
-  implicit def coderSparkeyURI: Coder[SparkeyUri] = Coder.kryo[SparkeyUri]
 }
 
-private class LocalSparkeyUri(val basePath: String) extends SparkeyUri {
+private case class LocalSparkeyUri(basePath: String) extends SparkeyUri {
   override def getReader: SparkeyReader =
     new ThreadLocalSparkeyReader(new File(basePath))
   override private[sparkey] def exists: Boolean =
     SparkeyUri.extensions.map(e => new File(basePath + e)).exists(_.exists)
 }
 
-private class RemoteSparkeyUri(val basePath: String, options: PipelineOptions) extends SparkeyUri {
+private case class RemoteSparkeyUri(basePath: String, rfu: RemoteFileUtil) extends SparkeyUri {
 
-  val rfu: RemoteFileUtil = RemoteFileUtil.create(options)
+  def this(basePath: String, options: PipelineOptions) =
```

Comment:
since this is a `case class` i would prefer this to be moved to a companion object `apply(basePath: String, options: PipelineOptions)`